### PR TITLE
fix(typeahead): prevent validation during an item click

### LINKED ISSUE #2719: NgbTypeahead required validation with reactive form works incorrectly
### Bug description:

In markup I have this code 

```html
<div class="first-group-margin form-group">
      <label for="recipient">{{ 'To' | translate }}:</label>
      <input id="recipient" type="text" class="tatoo-form-control" name="recipient" [ngbTypeahead]="search"
        [resultFormatter]="autocompleteFormatter" [inputFormatter]="autocompleteFormatter" [class.is-invalid]="recipient.invalid && recipient.touched"
        formControlName="recipient" [editable]="false" placeholder="{{ 'Placeholder.UserRecipient' | translate }}"
        autocomplete="false" (selectItem)="changeEvent($event)">
      <div class="alert-absolute alert alert-danger mt-2" *ngIf="recipient.touched && recipient.invalid">
        <span *ngIf="recipient.errors['required']">{{'Validation.BlankRecipient' | translate}}</span>
      </div>
</div>
```

and when I select value from dropdown the input field is 'blinking red' because change event occures earlier then typeahead select item event I think. So my form control become invalid for a while

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 6.0.2

ng-bootstrap: 2.0.0

Bootstrap: 4.1.1

### PR DESCRIPTION
**Pull request by fbasso**
fix #2719 

Landed here the solution proposed in #2843 (thanks @wannadream !), with some changes:

- The fix for this particular issue is separated from the others,
- The 'if' inside the handleBlur has been simplified,
- Protractor tests have been added to cover this case

### CODE REVIEW

**Comment by fbasso** [comment]
The fix of this issue has been done by #2869, so this PR is just a test to check it !

**Comment by benouat** [comment]
LGTM, even though I am not a big fan of this kind of test (ie playing with mousedown/mouseup/click).

Nonetheless, I'll let @maxokorokov decide about this PR, as I think he tried very hard to write a UT for that with no outcome other than _nope, it can't be done with a unit test_ 😆

File: e2e-app/src/app/typeahead/validation/typeahead-validation.component.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    </div>
+    <div class="input-group mr-2">
+      <label for="typeahead" class="mr-2">Typeahead</label>
+      <input
+        id="typeahead"
+        name="typeahead"
+        type="text"
+        class="form-control"
+        formControlName="field"
+        [class.is-invalid]="field.invalid && field.touched"
```
Comment (maxokorokov) [inline e2e-app/src/app/typeahead/validation/typeahead-validation.component.html]: forms will add `ng-touched/ng-untouched` class automatically

File: e2e-app/src/app/typeahead/validation/typeahead-validation.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  const expectValid =
+      async() => { expect(page.getErrorMsgElement().isPresent()).toBeFalsy(`The input should be valid`); };
+
+  beforeAll(() => page = new TypeaheadPage());
+
+  beforeEach(async() => await openUrl('typeahead/validation'));
+
+  it(`should stay valid on item click`, async() => {
+    await page.getTypeaheadInput().click();
```
Comment (maxokorokov) [inline e2e-app/src/app/typeahead/validation/typeahead-validation.e2e-spec.ts]: just click should be enough if I'm not mistaken?

File: e2e-app/src/app/typeahead/typeahead.po.ts [hunk 3]
```diff
@@ -9,5 +9,7 @@ export class TypeaheadPage {
 
   getDropdownItems() { return this.getDropdown().$$('button'); }
 
+  getErrorMsgElement() { return $('#error_msg'); }
```
Comment (maxokorokov) [inline e2e-app/src/app/typeahead/typeahead.po.ts]: not used antymore

File: e2e-app/src/app/typeahead/validation/typeahead-validation.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import {TypeaheadPage} from '../typeahead.po';
+import {Key, browser} from 'protractor';
+import {expectFocused, openUrl, sendKey} from '../../tools.po';
+
+fdescribe('Typeahead', () => {
```
Comment (maxokorokov) [inline e2e-app/src/app/typeahead/validation/typeahead-validation.e2e-spec.ts]: `fdescribe` → please remove focus

My fault, will open a separate PR to break the build in future cases

File: e2e-app/src/app/typeahead/validation/typeahead-validation.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import {TypeaheadPage} from '../typeahead.po';
+import {Key, browser} from 'protractor';
```
Comment (maxokorokov) [inline e2e-app/src/app/typeahead/validation/typeahead-validation.e2e-spec.ts]: Many unused imports

**Comment by fbasso** [comment]
Thanks, it's done !

**Pull request merged**